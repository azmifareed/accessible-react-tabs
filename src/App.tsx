import React from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme'
import { Tabs, Tab } from './components/Tabs';
import { GlobalStyle, AppContainer } from './App.styled';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle/>
      <AppContainer>
        <Tabs ariaLabel="Awesome Tabs" paramName="awesome-tab">
          <Tab
            id="tincidunt"
            label="Tincidunt"
            isSelected={false}
          >
            <h3>Donec vitae</h3>
            <p>Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. 
            Vivamus quis mi. Nullam vel sem. Morbi ac felis.</p>
            <p>Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, 
              ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. 
              In dui magna, posuere eget, vestibulum et, tempor auctor, justo.</p>
          </Tab>
          <Tab
            id="sodales"
            label="Sodales"
            isSelected={false}
          >
            <h3>Donec sodales sagittis</h3>
            <p>Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, 
              ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. 
              In dui magna, posuere eget, vestibulum et, tempor auctor, justo.</p>
            <p>Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. 
            Vivamus quis mi. Nullam vel sem. Morbi ac felis.</p>
          </Tab>
        </Tabs>

        <hr/>
        
        <Tabs ariaLabel="Another Tabs" paramName="another-tab">
          <Tab
            id="sodales"
            label="Sodales"
            isSelected={false}
          >
            <h3>Donec sodales sagittis</h3>
            <p>Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, 
              ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. 
              In dui magna, posuere eget, vestibulum et, tempor auctor, justo.</p>
            <p>Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. 
            Vivamus quis mi. Nullam vel sem. Morbi ac felis.</p>
          </Tab>
          <Tab
            id="tincidunt"
            label="Tincidunt"
            isSelected={false}
          >
            <h3>Donec vitae</h3>
            <p>Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. 
            Vivamus quis mi. Nullam vel sem. Morbi ac felis.</p>
            <p>Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, 
              ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. 
              In dui magna, posuere eget, vestibulum et, tempor auctor, justo.</p>
          </Tab>
        </Tabs>
      </AppContainer>
    </ThemeProvider>
  );
}

export default App;
