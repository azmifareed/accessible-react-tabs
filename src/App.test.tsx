import React from 'react';
import { render } from './utils/test-utils';
import App from './App';

test('renders app correctly', () => {
  const component = render(<App />);
  expect(component.baseElement).toMatchSnapshot();
});
