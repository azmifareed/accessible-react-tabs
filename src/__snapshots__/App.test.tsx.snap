// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app correctly 1`] = `
<body>
  <div>
    <div
      class="sc-eCApnc gntAJ"
    >
      <div
        class="sc-hKFxyN bPCfgy"
      >
        <div
          aria-label="Awesome Tabs"
          class="sc-bdnxRM ewiLTn"
          role="tablist"
        >
          <button
            aria-controls="tincidunt-tab"
            aria-selected="true"
            class="sc-gtsrHT hLdcqK"
            id="tincidunt"
            role="tab"
          >
            Tincidunt
          </button>
          <button
            aria-controls="sodales-tab"
            aria-selected="false"
            class="sc-gtsrHT crbYRc"
            id="sodales"
            role="tab"
            tabindex="-1"
          >
            Sodales
          </button>
        </div>
        <div
          aria-labelledby="tincidunt"
          class="sc-dlnjwi cIarac"
          id="tincidunt-tab"
          role="tabpanel"
          tabindex="0"
        >
          <h3>
            Donec vitae
          </h3>
          <p>
            Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. Vivamus quis mi. Nullam vel sem. Morbi ac felis.
          </p>
          <p>
            Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
          </p>
        </div>
        <div
          aria-labelledby="sodales"
          class="sc-dlnjwi dBvEUx"
          id="sodales-tab"
          role="tabpanel"
          tabindex="0"
        >
          <h3>
            Donec sodales sagittis
          </h3>
          <p>
            Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
          </p>
          <p>
            Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. Vivamus quis mi. Nullam vel sem. Morbi ac felis.
          </p>
        </div>
      </div>
      <hr />
      <div
        class="sc-hKFxyN bPCfgy"
      >
        <div
          aria-label="Another Tabs"
          class="sc-bdnxRM ewiLTn"
          role="tablist"
        >
          <button
            aria-controls="sodales-tab"
            aria-selected="false"
            class="sc-gtsrHT crbYRc"
            id="sodales"
            role="tab"
            tabindex="-1"
          >
            Sodales
          </button>
          <button
            aria-controls="tincidunt-tab"
            aria-selected="true"
            class="sc-gtsrHT hLdcqK"
            id="tincidunt"
            role="tab"
          >
            Tincidunt
          </button>
        </div>
        <div
          aria-labelledby="sodales"
          class="sc-dlnjwi dBvEUx"
          id="sodales-tab"
          role="tabpanel"
          tabindex="0"
        >
          <h3>
            Donec sodales sagittis
          </h3>
          <p>
            Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
          </p>
          <p>
            Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. Vivamus quis mi. Nullam vel sem. Morbi ac felis.
          </p>
        </div>
        <div
          aria-labelledby="tincidunt"
          class="sc-dlnjwi cIarac"
          id="tincidunt-tab"
          role="tabpanel"
          tabindex="0"
        >
          <h3>
            Donec vitae
          </h3>
          <p>
            Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. Vivamus quis mi. Nullam vel sem. Morbi ac felis.
          </p>
          <p>
            Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
